### PR TITLE
feat: show pancard status using unofficial APIs from Income Tax

### LINKED ISSUE #1701: Pancard Status
### Is your feature request related to a problem? Please describe.

We are having a difficulty with TDS returns because if a supplier and customer does not link their PAN card with his Aadhar, the government issues a notification in the deductor's name.(in our name) We would request you to know the status of the Suppliers and Customers Pancard, similar to how the GSTIN status is displayed in the supplier & customer.

image

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Sanket322**
closes: #1701

<sub><a href="https://huly.app/guest/resilienttech?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJsaW5rSWQiOiI2NmE4Nzg4ODdmMDllZGM1NjA2MjJjOTAiLCJndWVzdCI6InRydWUiLCJlbWFpbCI6IiNndWVzdEBoYy5lbmdpbmVlcmluZyIsIndvcmtzcGFjZSI6Inctc21pdHZvcmEyMDMtcmVzaWxpZW50dGVjLTY2N2U0MjkxLWEwNWMwNjY4N2EtNjM4MjY3IiwicHJvZHVjdElkIjoiIn0.GCadmvVWv_4QHnqdZ8p_BbiIn6dtSWcbsOljJ8BxEiE">Huly&reg;: <b>IC-2600</b></a></sub>

### CODE REVIEW

File: india_compliance/gst_india/overrides/party.py [hunk 1]
```diff
@@ -164,3 +166,28 @@ def make_address(doc):
...
+        "preLoginFlag": "Y",
+        "serviceName": "linkAadhaarPreLoginService",
+    }
+    response = requests.post(url, json=payload)
+    messages = response.json().get("messages")
+    if not messages:
+        return ""
+
+    error_code_desc_map = {
+        "EF40124": "Linked",  # pan linked to given aadhar card
```
Comment (vorasmit) [inline india_compliance/gst_india/overrides/party.py]: - [x] PAN Doctype: Created only if a status is available
- [x] Random Aadhar (Check performance)
- [x] Aadhar Invalid Error
- [x] Try Except: throw error with force refresh
- [x] PAN Info in utils
- [x] Status from Linked to Valid
- [x] If status is available: No refresh button on the face of doc. However user can refresh after going to PAN doc.

File: india_compliance/gst_india/client_scripts/party.js [hunk 2]
```diff
@@ -123,9 +123,18 @@ function set_gstin_options_and_status(doctype) {
         refresh(frm) {
             set_gstin_options(frm);
             india_compliance.set_gstin_status(frm.get_field("gstin"));
+            if (doctype !== "Address") {
+                india_compliance.set_pan_status(frm.get_field("pan"));
+            }
         },
         gstin(frm) {
             india_compliance.set_gstin_status(frm.get_field("gstin"));
+            if (doctype !== "Address") {
```
Comment (ljain112) [inline india_compliance/gst_india/client_scripts/party.js]: .

File: india_compliance/gst_india/doctype/pan/pan.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+// Copyright (c) 2024, Resilient Tech and contributors
+// For license information, please see license.txt
+
+frappe.ui.form.on("Pan", {
+    refresh(frm) {
+        frm.add_custom_button(__("Refresh PAN Status"), () => {
+            frm.call("update_pan_status");
+        });
+    },
+    pan(frm) {
```
Comment (ljain112) [inline india_compliance/gst_india/doctype/pan/pan.js]: backend

File: india_compliance/gst_india/overrides/party.py [hunk 4]
```diff
@@ -164,3 +168,119 @@ def make_address(doc):
             "is_shipping_address": doc.get("is_shipping_address"),
         }
     ).insert()
+
+
+multiplication_table = [
```
Comment (ljain112) [inline india_compliance/gst_india/overrides/party.py]: move to pan.py

File: india_compliance/gst_india/doctype/pan/pan.json [hunk 5]
```diff
@@ -0,0 +1,52 @@
+{
+    "actions": [],
+    "allow_rename": 1,
+    "creation": "2024-07-31 13:10:35.277697",
+    "doctype": "DocType",
+    "engine": "InnoDB",
+    "field_order": [
+        "pan",
+        "pan_status"
```
Comment (vorasmit) [inline india_compliance/gst_india/doctype/pan/pan.json]: field last updated on is missing

**Pull request merged**